avm/res/sql/server/elastic-pool/main.bicep

```diff
@@ -90,3 +102,94 @@ output resourceGroupName string = resourceGroup().name
 
 @description('The location the resource was deployed into.')
 output location string = elasticPool.location
+
+// =============== //
+//   Definitions   //
+// =============== //
+
+@export()
+@description('The per database settings for the elastic pool.')
+type elasticPoolPerDatabaseSettingsType = {
+  @description('Optional. Auto Pause Delay for per database within pool.')
+  autoPauseDelay: int?
+
+  @description('Reqired. The maximum capacity any one database can consume. Examples: \'0.5\', \'2\'.')
+  maxCapacity: string
+
+  // using string as minCapacity can be fractional
+  @description('Required. The minimum capacity all databases are guaranteed. Examples: \'0.5\', \'1\'.')
+  minCapacity: string
+}
+
+@export()
+@description('The elastic pool SKU.')
+type elasticPoolSkuType = {
+  @description('Optional. The capacity of the particular SKU.')
+  capacity: int?
+
+  @description('Optional. If the service has different generations of hardware, for the same SKU, then that can be captured here.')
+  family: string?
+
+  @description('Required. The name of the SKU, typically, a letter + Number code, e.g. P3.')
+  name:
+    | 'BasicPool'
+    | 'StandardPool'
+    | 'PremiumPool'
+    | 'GP_Gen5'
+    | 'GP_DC'
+    | 'GP_FSv2'
+    | 'BC_Gen5'
+    | 'BC_DC'
+    | 'HS_Gen5'
+    | 'HS_PRMS'
+    | 'HS_MOPRMS'
+    | 'ServerlessPool'
+
+  @description('Optional. Size of the particular SKU.')
+  size: string?
+
+  @description('Required. The tier or edition of the particular SKU, e.g. Basic, Premium.')
+  tier: string?
+}
+
+@export()
+type elasticPoolPropertyType = {
```

Comment:
I'd recommend to move this type (including the export, don't remove that 😄) into the parent module